File: LinkedList/LinkedList-Java/src/LinkedList/LinkedList.java
package LinkedList;

import java.time.temporal.IsoFields;

public class LinkedList 
{
  public LinkedList()
  {
    this.rear=null;
    this.front=null;
    this.size=0;
  }

  public void insertFront(String word)
  {
    Node newNode=new Node(new Data(word));
    if(isEmpty())
    {
      this.rear=newNode;
      this.front=newNode;
    }
    else 
    {
      newNode.setNext(front);
      this.front=newNode;
    }
    this.size++;
  }

  public void insertBack(String word)
  {
    Node newNode=new Node(new Data(word));
    if(isEmpty())
    {
      this.rear=newNode;
      this.front=newNode;
    }
    else 
    {
      this.rear.setNext(newNode);
      this.rear=newNode;
    }
    this.size++;
  }

  public void deleteFront()
  {
    if(isEmpty())
    {
      System.out.println("The List is empty");
      return;
    }

    this.front=this.front.getNext();
    if(this.front==null)
    {
      this.rear=null;
    }
    this.size--;
  }

  public void deleteRear()
  {
    if(isEmpty())
    {
      System.out.println("The List is Empty");
      return;
    }
     if(this.front==this.rear)
     {
       this.rear=null;
       this.front=null;
       this.size--;
       return;
     }
     Node current=this.front.getNext();
     while(current.getNext()!=this.rear)
     {
       current=current.getNext();
     }
     current.setNext(null);
     this.rear=current;
     this.size--;
  }

  public void deleteAt(int index)
  {
    if(index==0)
    {
      this.deleteFront();
      return;
    }
    else if(index==this.size)
    {
      this.deleteRear();
      return;
    }
    int count=0;
    Node current=this.front;
    while(count<index-1)
    {
      current=current.getNext();
      count++;
    }
    current.setNext(current.getNext().getNext());
    size--;
  }

  public void insertAt(String word,int index)
  {
    if(index<0||index>this.size)
    {
      return;
    }
    if(index==0)
    {
      this.insertFront(word);
      return;
    }
    if(index==this.size)
    {
      this.insertBack(word);
      return;
    }
    int count=0;
    Node newNode=new Node(new Data(word));
    Node current=this.front;
    while(count<index-1)
    {
      current=current.getNext();
      count++;
    }
    Node temp=current.getNext();
    current.setNext(newNode);
    newNode.setNext(temp);
    this.size++;
  }

  public void view()
  {
    if(isEmpty())
    {
      System.out.println("The List is empty");
      return;
    }

    Node current=this.front;
    while(current!=null)
    {
      System.out.println(current.getData().getWord());
      current=current.getNext();
    }
  }

  public int getSize()
  {
    return this.size;
  }

  private boolean isEmpty()
  {
    return this.front==null;
  }

  private Node rear;
  private Node front;
  private int size;
}

File: LinkedList/LinkedList-Python/src/LinkedList.py
from Node import Node
from Data import Data

class Link:
    def __init__(self):
        self.top=None
        self.bottom=None
        self.size=0

    def insertTop(self,data):
        newNode=Node(Data(data));
        if self.isEmpty():
            self.top=newNode
            self.bottom=newNode
        else:
            newNode.next=self.top
            self.top=newNode
        self.size+=1

    def insertBottom(self,data):
        newNode=Node(Data(data));
        if self.isEmpty():
            self.top=newNode
            self.bottom=newNode
        else:
            self.bottom.next=newNode
            self.bottom=newNode
        self.size+=1

    def insertAt(self,index,data):
        if index==0:
            self.insertTop(data)
            return
        elif index==self.size:
            self.insertBottom(data)
            return

        current=self.top
        newNode=Node(Data(data));
        count=0
        while count<index-1:
            current=current.next
            count+=1

        temp=current.next
        current.next=newNode
        newNode.next=current
        self.size+=1

    def deleteTop(self):
        if self.isEmpty():
            return

        self.top=top.next
        if top is None:
            self.bottom=None
        self.size-=1

    def deleteBottom(self):
        if self.isEmpty():
            return

        if  self.top==self.bottom:
            self.bottom=None
            self.top=None
            self.size-=1
            return

        count=0
        current=self.top
        while current!=self.bottom:
            current=current.next
            count+=1
        current.next=None
        self.bottom=current

    def deleteAt(self,index):
        if index<0 or index>self.size:
            return

        if index-1==0:
            self.deleteTop()
            return
        elif index-1==self.size:
            self.deleteBottom()
            return

        count=0
        current=self.top
        while count<index-1:
            current=current.next
            count+=1

        current.next=current.next.next
        self.size-=1

    def view(self):
        if self.isEmpty():
            return

        current=self.top
        while current!=None:
            print(current.data.word)
            current=current.next

    def isEmpty(self):
        return self.top is None
